src/commands/transaction/construct_transaction/add_action_1/add_action/use_global_contract/mod.rs

```diff
@@ -0,0 +1,121 @@
+    ))]
+    /// Deploy code as a global contract code hash (immutable)
+    AsGlobalHash(UseHashAction),
+    #[strum_discriminants(strum(
+        message = "as-global-account-id - Deploy code as a global contract account ID (mutable)"
+    ))]
+    /// Deploy code as a global contract account ID (mutable)
+    AsGlobalAccountId(UseAccountIdAction),
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = super::super::super::ConstructTransactionContext)]
+#[interactive_clap(output_context = UseHashActionContext)]
+pub struct UseHashAction {
+    /// What is a hash of the global contract?
+    pub hash: crate::types::crypto_hash::CryptoHash,
+    #[interactive_clap(subcommand)]
+    initialize: super::deploy_contract::initialize_mode::InitializeMode,
+}
+
+#[derive(Debug, Clone)]
+pub struct UseHashActionContext(super::super::super::ConstructTransactionContext);
+
+impl UseHashActionContext {
+    pub fn from_previous_context(
+        previous_context: super::super::super::ConstructTransactionContext,
+        scope: &<UseHashAction as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let action = near_primitives::transaction::Action::UseGlobalContract(Box::new(
+            near_primitives::action::UseGlobalContractAction {
+                contract_identifier: near_primitives::action::GlobalContractIdentifier::CodeHash(
+                    scope.hash.into(),
+                ),
+            },
+        ));
+        let mut actions = previous_context.actions;
+        actions.push(action);
+        Ok(Self(super::super::super::ConstructTransactionContext {
+            global_context: previous_context.global_context,
+            signer_account_id: previous_context.signer_account_id,
+            receiver_account_id: previous_context.receiver_account_id,
+            actions,
+        }))
+    }
+}
+impl From<UseHashActionContext> for super::super::super::ConstructTransactionContext {
+    fn from(item: UseHashActionContext) -> Self {
+        item.0
+    }
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = super::super::super::ConstructTransactionContext)]
+#[interactive_clap(output_context = UseAccountIdActionContext)]
+pub struct UseAccountIdAction {
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is a account ID of the global contract?
+    pub account_id: crate::types::account_id::AccountId,
+    #[interactive_clap(subcommand)]
+    initialize: super::deploy_contract::initialize_mode::InitializeMode,
+}
+
+impl UseAccountIdAction {
+    pub fn input_account_id(
+        context: &super::super::super::ConstructTransactionContext,
+    ) -> color_eyre::eyre::Result<Option<crate::types::account_id::AccountId>> {
+        crate::common::input_signer_account_id_from_used_account_list(
```

Comment:
nit: looks like `input_non_signer_account_id_from_used_account_list` fits as used global account generally 
doesn't have to be a signer of current or of any of any of previous transactions